src/platform/qt/library/LibraryTree.h

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2014-2017 waddlesplash
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#pragma once
+
+#include <QTreeWidget>
+
+#include "LibraryController.h"
+
+// Predefinitions
+namespace Ui { class LibraryTree; }
+
+namespace QGBA {
+
+class LibraryTree : public QTreeWidget, public AbstractGameList {
+Q_OBJECT
+
+public:
+	explicit LibraryTree(QWidget* parent_widget = nullptr);
+	~LibraryTree();
+
+	// AbstractGameList stuff
+	virtual LibraryEntry selectedEntry() override;
+	virtual void selectEntry(LibraryEntry game) override;
+
+	virtual void setViewStyle(LibraryStyle newStyle) override;
+
+	virtual void addEntries(QList<LibraryEntry> items) override;
+	virtual void addEntry(LibraryEntry item) override;
+	virtual void removeEntry(LibraryEntry item) override;
+
+signals:
+	void StartGame();
+
+private slots:
+	void ItemActivated(QTreeWidgetItem* item);
+
+private:
+	enum Columns {
```

Comment:
These are the indexes of the columns as they are in the QTreeWidget. So if we insert/add/remove/etc. columns, the numbers will change, and rather than changing a bunch of magic numbers all over the .cpp, instead you just change this enum. Keeps everything nice and tidy.